core/src/main/java/org/infinispan/util/concurrent/locks/impl/DefaultPendingLockManager.java

```diff
@@ -77,14 +83,14 @@ public PendingLockPromise checkPendingTransactionsForKey(TxInvocationContext<?>
          return createAndStore(getTransactionWithLockedKey(txTopologyId, key, globalTransaction),
                                globalTransaction, time, unit);
       }
-      return PendingLockPromise.NO_OP;
+      return createAndStoreDefault(globalTransaction, time, unit);
```

Comment:
I was trying to solve a deadlock by avoiding the remote command to double check with the DefaultPendingLockManager. However, I've managed to solve it in other way and I think this code can be removed now.